### ISSUE TITLE
Crash when hiding the container view which contains PagingViewController with Animation

### ISSUE THREAD

**Issue by ankit0812** [rechsteiner/Parchment#633]
Console Logs :

```
libc++abi: terminating with uncaught exception of type NSException
dyld4 config: DYLD_LIBRARY_PATH=/usr/lib/system/introspection DYLD_INSERT_LIBRARIES=/Developer/usr/lib/libBacktraceRecording.dylib:/Developer/usr/lib/libMainThreadChecker.dylib:/Developer/Library/PrivateFrameworks/DTDDISupport.framework/libViewDebuggerSupport.dylib
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'UICollectionViewLayoutAttributes: -setFrame: with CGRectNull is undefined.'
terminating with uncaught exception of type NSException
```
Thread 1: "UICollectionViewLayoutAttributes: -setFrame: with CGRectNull is undefined."

Let me know if you need more details

Stack Trace :
<img width="579" alt="Screenshot 2022-01-27 at 2 57 53 PM" src="https://user-images.githubusercontent.com/21132111/151330765-5015638a-8809-414a-b649-07c317704322.png">

**Comment by ankit0812** [comment]
```
*** Terminating app due to uncaught exception 'CALayerInvalidGeometry', reason: 'CALayer bounds contains NaN: [nan 0; 0 44]. Layer: <CALayer:0x28079d580; position = CGPoint (0 22); bounds = CGRect (0 0; 0 44); delegate = <UICollectionView: 0x11ea8b400; frame = (0 0; 0 44); clipsToBounds = YES; gestureRecognizers = <NSArray: 0x280f35380>; animations = { bounds.origin=<CABasicAnimation: 0x2804e48a0>; bounds.size=<CABasicAnimation: 0x2804e4f20>; position=<CABasicAnimation: 0x2804e5260>; }; layer = <CALayer: 0x28079d580>; contentOffset: {0, 0}; contentSize: {210.33333333333331, 44}; adjustedContentInset: {0, 0, 0, 0}; layout: <Parchment.PagingCollectionViewLayout: 0x125692d10>; dataSource: <Parchment.PagingController: 0x12568bb90>>; sublayers = (<CALayer: 0x280462e80>, <CALayer: 0x280462f60>, <CALayer: 0x280462e60>, <CALayer: 0x2804603c0>); masksToBounds = YES; allowsGroupOpacity = YES; backgroundColor = <CGColor 0x282319c20> [<CGColorSpace 0x282447000> (kCGColorSpaceICCBased; kCGColorSpaceModelMonochrome; Generic Gray Gamma 2.2 Profile; extended range)] ( 1 1 ); animations = [bounds.origin=<CABasicAnimation: 0x2804e48a0> bounds.size=<CABasicAnimation: 0x2804e4f20> position=<CABasicAnimation: 0x2804e5260>]>'
terminating with uncaught exception of type NSException
```

On some cases it crashes here. Kindly add exception handling for the same.

Stack Trace :

<img width="870" alt="Screenshot 2022-01-27 at 5 27 34 PM" src="https://user-images.githubusercontent.com/21132111/151354651-77a07cd0-f234-4385-89dd-1dd21d993548.png">

**Comment by ankit0812** [comment]
@rechsteiner Please help on this?

**Comment by kf99916** [comment]
I also encounter `CALayerInvalidGeometry ` issue in `PagingController.updateScrollingState(pagingItem:upcomingPaging|tem:initialContentOffset:distance:progress:)`

Version: 3.2.0

```
Fatal Exception: CALayerInvalidGeometry
CALayer bounds contains NaN: [nan 0; 0 48]. Layer: <CALayer:0x283bda1a0; position = CGPoint (0 24); bounds = CGRect (0 0; 0 48); delegate = <UICollectionView: 0x155b36200; frame = (0 0; 0 48); clipsToBounds = YES; gestureRecognizers = <NSArray: 0x2831f9fb0>; layer = <CALayer: 0x283bda1a0>; contentOffset: {0, 0}; contentSize: {375, 48}; adjustedContentInset: {0, 0, 0, 0}; layout: <Parchment.PagingCollectionViewLayout: 0x156a73850>; dataSource: <Parchment.PagingController: 0x156896080>>; sublayers = (<CALayer: 0x28386e320>, <CALayer: 0x28386d980>, <CALayer: 0x28386e140>, <CALayer: 0x28386d660>, <CALayer: 0x28386ea20>, <CALayer: 0x283861900>); masksToBounds = YES; allowsGroupOpacity = YES; backgroundColor = <CGColor 0x281cf3020> [<CGColorSpace 0x281bcfb40> (kCGColorSpaceICCBased; kCGColorSpaceModelMonochrome; 通用灰階光度2.2色彩描述; extended range)] ( 1 1 )>
```

**Comment by rechsteiner** [comment]
Hey! Sorry for the late response. Do you have any steps on how to reproduce this?

**Comment by rechsteiner** [comment]
This should be fixed now with #680. It's available in the 4.0 branch and will be released soon. Please let me know if you're still seeing this issue and I'll re-open this.

**Comment by thangpq264** [comment]
@rechsteiner When will the 4.0 branch be released?
